tests/test_ungriddeddata.py

```diff
@@ -239,3 +240,56 @@ def test_from_single_station_data():
     data0 = stat.ec550aer
     data1 = d.all_datapoints_var("ec550aer")
     assert data0 == pytest.approx(data1, abs=1e-20)
+
+
+def test_last_meta_idx(aeronetsunv3lev2_subset: UngriddedData):
+    assert isinstance(aeronetsunv3lev2_subset.last_meta_idx, (np.ndarray, np.generic))
+
+
+def test_has_flag_data(aeronetsunv3lev2_subset: UngriddedData):
+    assert isinstance(aeronetsunv3lev2_subset.has_flag_data, (np.bool_, bool))
+
+
+def test_is_filtered(aeronetsunv3lev2_subset: UngriddedData):
+    assert isinstance(aeronetsunv3lev2_subset.is_filtered, (np.bool_, bool))
+
+
+def test_available_meta_keys(aeronetsunv3lev2_subset: UngriddedData):
+    assert isinstance(aeronetsunv3lev2_subset.available_meta_keys, list)
+    assert np.all(isinstance(key, str) for key in aeronetsunv3lev2_subset.available_meta_keys)
+
+
+def test_nonunique_station_names(aeronetsunv3lev2_subset: UngriddedData):
+    assert isinstance(aeronetsunv3lev2_subset.nonunique_station_names, list)
+
+
+def test_set_flags_nan_error(aeronetsunv3lev2_subset: UngriddedData):
+    data = aeronetsunv3lev2_subset.copy()
+    with pytest.raises(AttributeError) as e:
+        data = data.data.set_flags_nan(inplace=True)
+    assert e.type is AttributeError
+
+
+def test_remove_outliers(aeronetsunv3lev2_subset: UngriddedData):
+    data = aeronetsunv3lev2_subset.copy()
+    assert not data.filter_hist
+    new = data.remove_outliers(var_name="od550aer", low=0, high=0)
+    assert new.filter_hist
+
+
+def test_extract_var(aeronetsunv3lev2_subset: UngriddedData):
+    data = aeronetsunv3lev2_subset.copy()
+    od = data.extract_var("od550aer")
+    assert not data.is_filtered
+    assert od.is_filtered
+    assert od.shape[0] < data.shape[0]
+    with pytest.raises(VariableDefinitionError) as e:
+        data.extract_var("nope")
+    assert e.type is VariableDefinitionError
+
+
+def test_find_common_stations(aeronetsunv3lev2_subset: UngriddedData):
+    data1 = aeronetsunv3lev2_subset.copy()
+    data2 = aeronetsunv3lev2_subset.copy()
+    station_map = data1.find_common_stations(other=data2)
+    assert np.all(key == station_map[key] for key in station_map)
```

Comment:
why `np.all` instead of plain `all`?
maybe it could be written as:
``` python
    assert station_map == {key: key for key in station_map)
```
or 
``` python
    assert station_map.keys() == station_map.values()
```